### DISCUSSION TITLE
tutorials/export/changing_application_icon_for_windows

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#245] [User-contributed notes]
# tutorials/export/changing_application_icon_for_windows

By default, the exported project's icon will be the Godot icon. You will most likely want to change that for your project. There are two types of icons that can be changed on Windows: the file icon...

https://docs.godotengine.org/en/stable/tutorials/export/changing_application_icon_for_windows.html

**Comment by Rokle** [comment]
If icon is blurry on high resolutions but normal on low try this auto resize order
```
magick convert icon.png -define icon:auto-resize=16,32,48,64,128,256 icon.ico
```

**Reply by Khyze** [reply]
Thanks! Altough it gives a warning to just use "magick" instead of "magick convert" due to being deprecated

**Reply by TehBrian** [reply]
For easy copy-pasting:

```
magick icon.png -define icon:auto-resize=16,32,48,64,128,256 icon.ico
```

**Comment by lietu** [comment]
Replacing the Windows .exe icon from a Linux build pipeline was quite challenging, but I finally managed to get it working with:

```bash
# Remove any previously set icons
xvfb-run "$WINE" "ResourceHacker.exe" \
  -open "${PLATFORM_EXECUTABLE}" \
  -save "${PLATFORM_EXECUTABLE}" \
  -action delete \
  -mask ICONGROUP

xvfb-run "$WINE" "ResourceHacker.exe" \
  -open "${PLATFORM_EXECUTABLE}" \
  -save "${PLATFORM_EXECUTABLE}" \
  -action delete \
  -mask MAINICON

# Try to set icon with ResourceHacker
xvfb-run "$WINE" "ResourceHacker.exe" \
  -open "${PLATFORM_EXECUTABLE}" \
  -save "${PLATFORM_EXECUTABLE}" \
  -action modify \
  -res icon.ico \
  -mask ICONGROUP,MAINICON,0
  
# Try to set icon with rcedit
xvfb-run "$WINE" "rcedit-x64.exe" "${PLATFORM_EXECUTABLE}" --set-icon icon.ico
```

Took me several hours to finally get something that works and at this point I'm not sure which of the steps is 100% necessary, but this combination seems to work and produce predictable results and that's good enough for me right now.

This depends on `xvfb-run` being installed, `wine64` being available, path to which set to `$WINE`, `rcedit-x64.exe` from rcedit GitHub releases and `ResourceHacker.exe` from their homepage being available in the directory with the game .exe (filename set to `$PLATFORM_EXECUTABLE`), and `icon.ico` with all the wanted resolutions for the icon, e.g. generated from a square SVG with:

```bash
magick -background none icon.svg -density 300 -define icon:auto-resize=256,128,96,80,72,64,48,32,24,16 icon.ico
```

**Comment by Draradech** [comment]
None of this is necessary (since 4.1) unless you need manual control over the specific rescaled versions.
The exporter gained ico creation support here: https://github.com/godotengine/godot/pull/68828 (Godot 4.0)
It gained fallback to application/config/icon here: https://github.com/godotengine/godot/pull/77821 (Godot 4.1)
For taskbar: this always had the fallback to application/config/icon (since native icon was introduced)

**Reply by Calinou** [reply]
Issue opened:

- https://github.com/godotengine/godot-docs/issues/10518

**Reply by Static-Conection-Studios** [reply]
Presumably the part about having to download recedit still applies thou?

**Reply by Calinou** [reply]
You still need rcedit until Godot 4.4.x, but it won't be needed anymore in 4.5 and later.

**Comment by sszigeti** [comment]
When using a Mac to export for Windows, you need to install Wine first (e.g. `brew install --cask --no-quarantine wine-stable`).

Then, download rcedit-x64.exe -- the most recent version from https://github.com/electron/rcedit/releases -- and save it somewhere on your system. (I've chosen `/User/<my username>/Applications`).

Then, open Godot's Editor Settings, and go to **Export >Windows**.

* Select the location of the `rcedit-x64.exe` for the `rcedit`˙field.
* Select the location of your Wine executable (e.g. `/opt/homebrew/bin/wine` -- you can check this with the `which wine` command) in the `wine` field.

Now the exported .exe will have the correct application icon, instead of the Godot default.

**Reply by nubels** [reply]
Thank you so much! I had wine set to ´/Applications/Wine Stable.app/Contents/MacOS/wine´ and I was wondering why it didn't work. Now it works :)

**Reply by metamagicshaman** [reply]
wow same here! I was going crazy.

**Comment by viktor-ferenczi** [comment]
Went through the process of setting up proper Windows export on Linux Mint with Godot 4.4. It was quick, fortunately:
- Download `rcedit-x64.exe` from https://github.com/electron/rcedit/releases
- Install wine from Software Manager (or using apt from the command line)
- Specify the absolute path to `rcedit-x64.exe` in Godot's Editor config (search for `rcedit`)

The only thing missing here is to prove the above instructions clearly inside the Export process somehow instead of a cryptic error message.

**Reply by Calinou** [reply]
> The only thing missing here is to prove the above instructions clearly inside the Export process somehow instead of a cryptic error message.

4.5 won't need rcedit anymore:

- https://github.com/godotengine/godot/pull/75950

There's a PR improving error reporting for 3.x still:

- https://github.com/godotengine/godot/pull/109498
